internal/bundle/utils_test.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package bundle
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/open-policy-agent/opa/internal/file/archive"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/util/test"
+)
+
+func TestGenerateBundleInfoWithFileDir(t *testing.T) {
+
+	files := map[string]string{
+		"/fuz/data.json":   "[1,2,3]",
+		"/data.json":       `{"a": {"b": {"c": [123]}}}`,
+		"/foo/policy.rego": "package foo\np = 1",
+		"/baz/authz.rego":  "package foo\nx = 1",
+		"base.rego":        "package bar\nx = 1",
+		"/.manifest":       `{"roots": ["foo", "bar", "fuz", "baz", "a"], "revision": "rev"}`,
+	}
+
+	test.WithTempFS(files, func(rootDir string) {
+		info := NewInfo(rootDir)
+
+		err := info.GenerateBundleInfo()
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		expectedManifest := bundle.Manifest{
+			Revision:      "rev",
+			Roots:         &[]string{"foo", "bar", "fuz", "baz", "a"},
+			WasmResolvers: nil,
+			Metadata:      nil,
+		}
+
+		if !info.Manifest.Equal(expectedManifest) {
+			t.Fatalf("expected manifest %v, but got: %v", expectedManifest, info.Manifest)
+		}
+
+		expectedPackages := map[string][]string{}
+		expectedPackages["data.bar"] = []string{filepath.Join(rootDir, "base.rego")}
+		expectedPackages["data.foo"] = []string{filepath.Join(rootDir, "baz/authz.rego"), filepath.Join(rootDir, "foo/policy.rego")}
+
+		if !reflect.DeepEqual(info.Packages, expectedPackages) {
+			t.Fatalf("expected packages %v, but got %v", expectedPackages, info.Packages)
+		}
+
+		expectedData := map[string]string{}
+		expectedData[""] = filepath.Join(rootDir, "data.json")
+		expectedData["data.fuz"] = filepath.Join(rootDir, "fuz/data.json")
```

Comment:
[style] It's largely a matter of preference, I guess:

```suggestion
		expectedData := map[string]string{
				"": filepath.Join(rootDir, "data.json"),
				"data.fuz": filepath.Join(rootDir, "fuz/data.json"),
		}
```

Reply:
Updated.